File: topic_5/Python/6-menores.py
# Recebe um numero n e uma lista e retorna os n menores elementos
# da lista na ordem que aparecem na lista

def menores(n, xs):
    if len(xs) == n:
        return xs
    
    if min(xs) == max(xs):
        return xs[len(xs)-n:len(xs)]
    else:
        return menores(n, [x for x in xs if x != max(xs)])

print(menores(0, [6,2,2,4,9]) == []) #True
print(menores(1, [6,2,2,4,9]) == [2]) #True
print(menores(3, [5,5,5]) == [5,5,5]) #True
print(menores(1, [5,5,5]) == [5]) #True
print(menores(3, [6,2,3,4,9]) == [2,3,4]) #True
print(menores(3, [5,3,1,9,7,2]) == [3,1,2]) #True

File: topic_5/Haskell/6-menores.hs
-- Recebe um numero n e uma lista e retorna os n menores elementos
-- da lista na ordem que aparecem na lista

menores 0 xs = []
menores n xs
    | length xs == n = xs
    | minimum xs == maximum xs = [(length xs-n)..(length xs)]
    | otherwise = menores n (filter (/= maximum xs) xs)

main = do
    a <- readLn :: IO Int
    b <- readLn :: IO [Int]
    print $ menores a b